윤건수/baekjoon/G2_16946/Main.java

```diff
@@ -0,0 +1,109 @@
+package baekjoon.G2_16946;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.*;
+
+public class Main {
+
+    static final int WALL = 1;
+    static final int EMPTY = 0;
+    static int[] dr = {1, -1, 0, 0};
+    static int[] dc = {0, 0, 1, -1};
+    static boolean[][] visited;
+
+    public static void main(String[] args) throws IOException {
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        int row = Integer.parseInt(st.nextToken());
+        int col = Integer.parseInt(st.nextToken());
+        int[][] map = new int[row][col];
+        visited = new boolean[row][col];
+        for(int i = 0; i < row; i++) {
+            String line = br.readLine();
+            for(int j = 0; j < col; j++) {
+                map[i][j] = Character.getNumericValue(line.charAt(j));
+            }
+        }
+
+        for(int i = 0; i < row; i++) {
+            for(int j = 0; j < col; j++) {
+                if(map[i][j] == EMPTY && !visited[i][j]){
+                    checkCount(map, i, j);
+                }
+            }
+        }
+
+        StringBuilder result = new StringBuilder();
+        for (int[] answerRow : map) {
+            for (int answerCol : answerRow) {
+                result.append(answerCol % 10);
+            }
+            result.append(System.lineSeparator());
+        }
+        result.deleteCharAt(result.length() - 1);
+        System.out.println(result);
+    }
+
+    private static void checkCount(int[][] map, int row, int col) {
+        visited[row][col] = true;
+        Queue<Point> que = new LinkedList<>();
+        que.add(new Point(row, col));
+
+        Set<Point> wallList = new HashSet<>();
+        int count = 1;
+        while(!que.isEmpty()){
+            Point curPoint = que.poll();
+            for (int i = 0; i < 4; i++) {
+                int nextRow = curPoint.row + dr[i];
+                int nextCol = curPoint.col + dc[i];
+                boolean isValidNext = isValid(visited, map, nextRow, nextCol);
+                if (isValidNext && map[nextRow][nextCol] == EMPTY
+                ) {
+                    count++;
+                    visited[nextRow][nextCol] = true;
+                    que.add(new Point(nextRow, nextCol));
+                } else if (isValidNext && map[nextRow][nextCol] != EMPTY
+                ) {
+                    wallList.add(new Point(nextRow, nextCol));
+                }
```

Comment:
조건이 복잡해 진다면 공통 조건은 따로 빼도 괜찮을 것 같네요.
정말 힘든 부분이 `depth 깊어짐 vs 조건이 복잡함` 인 것 같네요.

```suggestion
                if(isValidNext) {
                    if (map[nextRow][nextCol] == EMPTY) {
                        count++;
                        visited[nextRow][nextCol] = true;
                        que.add(new Point(nextRow, nextCol));
                    } else if (map[nextRow][nextCol] != EMPTY) {
                        wallList.add(new Point(nextRow, nextCol));
                    }
                }
```

Reply:
지금 케이스는 두 가지 표현다 나쁘지 않다고 생각합니다 ㅎㅎ
중첩 if 문을 줄이는 방식을 개인적으로 지향해서 해당 방식으로 하였습니다 
지금생각하니 
```
if(!isValidNext) continue;

if (map[nextRow][nextCol] == EMPTY) {
 // 내용
}else{
 // 내용
}
```

해버리는 방법도 있었겠네요

Reply:
저는 depth 줄이는 방법을 선호해서 말씀 드렸는데 제안해주신 방법이 너무 마음에 드네요.